tests/scripts/create-dev-cluster.sh

```diff
@@ -147,41 +173,56 @@ while getopts ":hrmfd:" opt; do
 	d)
 	    ROOK_EXAMPLES_DIR="$OPTARG"
 	    ;;
+    y)
+	    minikube_driver="$OPTARG"
+	    ;;
 	\?)
-	    echo  "Invalid option: -$OPTARG" >&2
+	    echo  "Invalid option: -$opt" >&2
 	    show_usage
 	    exit 1
 	    ;;
 	:)
-	    echo "Option -$OPTARG requires an argument." >&2
+	    echo "Option -$opt requires an argument." >&2
 	    exit 1
 	    ;;
     esac
 done
 
-echo "Using '$ROOK_EXAMPLES_DIR' as examples directory.."
+echo "Using '${ROOK_EXAMPLES_DIR}' as examples directory.."
 
-cd "$ROOK_EXAMPLES_DIR" || exit
+cd "${ROOK_EXAMPLES_DIR}" || exit
 check_examples_dir
 
-if [ -z "$force_minikube" ]; then
-    check_minikube_exists
+    check_minikube_profile_exists
+    exists=$?
+    if [[ $exists -eq 0 ]]; then
+if [[ -n "${force_minikube}" ]]; then
+    if [[ ${exists} -eq 0 ]]; then
+if [[ -z "${force_minikube}" ]]; then
+    check_minikube_profile_exists
+    exists=$?
+    if [[ ${exists} -eq 0 ]]; then
+    	setup_minikube_env
+    fi
 fi
```

Comment:
The current logic is: if we are not enforcing the creation (`-f` option) then we check if the profile exists otherwise we will just check if the profile exists in such case show an error. I don't see the need for changing the logic unless there's some good reason for it.

Reply:
@rkachach  wrote:
> The current logic is: if we are not enforcing the creation (`-f` option) then we check if the profile exists otherwise we will just check if the profile exists in such case show an error.

Indeed. That's what I wanted to change:
For me it was an unexpected behavior to error out on an existing env if not in force-create mode. My expectation was to use an existing env if not in force mode. Therefore, I think we should  to *always* check for existence and react differently on the result depending on whether we are in force mode. 




> I don't see the need for changing the logic unless there's some good reason for it.

My (IMHO good) reason was that the current logic lets the  script act in (for me) unexpected ways. I wanted to change that behavior

Reply:
@obnoxxx the idea of the script is to create everything from scratch so we are not trying to reuse the user env here. Specially with the PR (https://github.com/rook/rook/pull/13173) which adds support for creating multiple clusters where the user can specify the profile.